spec/index.md

```diff
@@ -613,6 +621,27 @@ Calls must stay in `replied` or `rejected` long enough for polling users to catc
 
 When asking the IC about the state or call of a request, the user uses the request id (see [Request ids](#request-id)) to read the request status (see [Request status](#state-tree-request-status)) from the state tree (see [Request: Read state](#http-read-state)).
 
+#### Synchronous canister calling {#http-sync-call}
+
+Unlike asynchronous canister calling, synchronous canister calling does not initially require the user to poll the Internet Computer for the status of the request. Instead, the user waits for a certified response from the Internet Computer for the initial call. However, if a certified response is not received (due to a timeout or pre-execution error), the user must poll the Internet Computer for the status. Still, the state transitions and semantics of the call itself are the same as for asynchronous canister calling.
```

Comment:
In case of a pre-execution error, polling makes no sense since the request status is not in the certified state tree (in such a case, the HTTP handler synchronously rejects the request).

Reply:
What if the replica is malicious? I.e. it sends a `reject` response back to the user, but actually submitted the ingress message.

Reply:
I rewrote the section and removed this part.